src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -163,15 +262,29 @@ void Instance::HandleSetCookingParameters(HandlerContext & ctx, const Commands::
     uint8_t reqCookMode;
     uint32_t reqCookTime;
     uint8_t reqPowerSetting;
-    auto & cookMode     = req.cookMode;
-    auto & cookTime     = req.cookTime;
-    auto & powerSetting = req.powerSetting;
+    uint8_t reqWattSettingIndex;
+    bool reqStartAfterSetting;
+    auto & cookMode          = req.cookMode;
+    auto & cookTime          = req.cookTime;
+    auto & powerSetting      = req.powerSetting;
+    auto & wattSettingIndex  = req.wattSettingIndex;
+    auto & startAfterSetting = req.startAfterSetting;
 
     opState = mOpStateInstance.GetCurrentOperationalState();
     VerifyOrExit(opState == to_underlying(OperationalStateEnum::kStopped), status = Status::InvalidInState);
 
-    VerifyOrExit(cookMode.HasValue() || cookTime.HasValue() || powerSetting.HasValue(), status = Status::InvalidCommand;
-                 ChipLogError(Zcl, "Microwave Oven Control: Failed to set cooking parameters, all command fields are missing "));
+    if (startAfterSetting.HasValue())
+    {
+        // TODO: ServerClusterCommandExists may always be returns true
```

Comment:
I don't understand this comment.  ServerClusterCommandExists in fact does the right thing, last I checked....

Reply:
Just I saw another overrided ServerClusterCommandExists is always returns true, I have tested and no problem. Never mind.

Reply:
@liangpy4 That's only compiled in unit tests that don't include the "real"  ServerClusterCommandExists.